.clang-tidy

```diff
@@ -0,0 +1,19 @@
+---
```

Comment:
These are the changes that are being made. It's a conservative set and could easily be expanded.
These are the options that are available:
```
    modernize-avoid-bind
    modernize-deprecated-headers
    modernize-loop-convert
    modernize-make-shared
    modernize-make-unique
    modernize-pass-by-value
    modernize-raw-string-literal
    modernize-redundant-void-arg
    modernize-replace-auto-ptr
    modernize-shrink-to-fit
    modernize-use-auto
    modernize-use-bool-literals
    modernize-use-default
    modernize-use-emplace
    modernize-use-nullptr
    modernize-use-override
    modernize-use-using
```
And they are all explained here: http://clang.llvm.org/extra/clang-tidy/checks/list.html